### PR TITLE
docs: update input/number

### PR DESCRIPTION
**Pull request by jasonren0403**
### Description

get rid of LiveSampleError and outdated content

### CODE REVIEW

File: files/zh-cn/web/html/element/input/number/index.md [hunk 1]
```diff
@@ -3,108 +3,160 @@ title: <input type="number">
...
+
+除了 {{HTMLElement("input")}} 元素共有的属性外，数字输入还提供以下属性。
+
+### `list`
+
+list 属性的值是位于同一文档中的 {{HTMLElement("datalist")}} 元素的 {{domxref("Element.id", "id")}}。{{HTMLElement("datalist")}} 提供了可输入到当前输入框的一个预定义的值列表。列表中任何与 {{htmlattrxref("type", "input")}} 不兼容的值都不包括在建议选项中。所提供的值是建议，不是要求：用户可以从这个预定义的列表中选择，或者提供不同的值。
+
+### `max`
+
+允许值范围内的最大值。如果输入到元素中的 {{htmlattrxref("value", "input")}} 超过此值，则元素将无法通过[约束验证](/zh-CN/docs/Web/Guide/HTML/HTML5/Constraint_validation)。如果 `max` 属性的值不是数字，则元素没有最大值。
```
Comment (jasonren0403) [inline files/zh-cn/web/html/element/input/number/index.md:65]:
```suggestion
允许值范围内的最大值。如果输入到元素中的 {{htmlattrxref("value", "input")}} 超过此值，则元素将无法通过[约束验证](/zh-CN/docs/Web/Guide/HTML/Constraint_validation)。如果 `max` 属性的值不是数字，则元素没有最大值。
```

File: files/zh-cn/web/html/element/input/number/index.md [hunk 2]
```diff
@@ -3,108 +3,160 @@ title: <input type="number">
...
+
+此值必须大于或等于 `min` 属性的值。
+
+### `min`
+
+允许值范围内的最小值。如果元素的 {{htmlattrxref("value", "input")}} 小于此值，则该元素将无法通过 [约束验证](/zh-CN/docs/Web/Guide/HTML/HTML5/Constraint_validation)。如果为 `min` 指定的值不是有效数字，则输入没有最小值。
```
Comment (jasonren0403) [inline files/zh-cn/web/html/element/input/number/index.md:71]:
```suggestion
允许值范围内的最小值。如果元素的 {{htmlattrxref("value", "input")}} 小于此值，则该元素将无法通过 [约束验证](/zh-CN/docs/Web/Guide/HTML/Constraint_validation)。如果为 `min` 指定的值不是有效数字，则输入没有最小值。
```

File: files/zh-cn/web/html/element/input/number/index.md [hunk 3]
```diff
@@ -325,9 +404,15 @@ switchBtn.addEventListener('click', function() {
...
 ```
 
-After declaring a few variables, we add an event listener to the button to control the switching mechanism. This is pretty simple, mostly involving changing over the button class and label, and updating the display values of the two sets of inputs when the button is pressed. Note that we're not converting back and forth between meters and feet/inches here, which a real-life web application would probably do.
+在声明了一些变量之后，我们为按钮添加一个事件监听器来控制切换机制。这很简单，主要是改变按钮的类别和标签，并在按钮被按下时更新两组输入的显示值。请注意，我们不是在米和英尺/英寸之间来回转换，现实生活中的网络应用可能会这么做。
+
+> **备注：** 请注意，当用户点击按钮时，我们从隐藏的输入中移除 `required` 属性，并清空 `value` 属性。这样我们就可以在两个输入组都没有填写的情况下提交表单，而不会提交那些我们不想提交的数据。如果我们不这样做，你就必须同时填入英尺/英寸**和**米来提交表单了！
+
+## 无障碍
+
+`<input type="number">` 元素的隐含 [role](/zh-CN/docs/Web/Accessibility/ARIA/Roles) 是 [`spinbutton`](/zh-CN/docs/Web/Accessibility/ARIA/Roles/spinbutton_role) 。如果 spinbutton 对你的表单控件来说不是一个重要的功能，那就*不要*使用 `type="number"`；此时，请使用带有 [`pattern`](/zh-CN/docs/Web/HTML/attributes/pattern) 属性的 [`inputmode="numeric"`](/zh-CN/docs/Web/HTML/Global_attributes/inputmode)，将字符限制在数字和相关字符。如果坚持使用 `<input type="number">`，用户在试图做其他事情时，有可能意外地增加一个数字。此外，如果用户试图输入不是数字的东西，也没有明确的反馈说明他们做错了什么。
```
Comment (jasonren0403) [inline files/zh-cn/web/html/element/input/number/index.md:413]:
```suggestion
`<input type="number">` 元素的隐含 [role](/zh-CN/docs/Web/Accessibility/ARIA/Roles) 是 [`spinbutton`](/zh-CN/docs/Web/Accessibility/ARIA/Roles/spinbutton_role) 。如果 spinbutton 对你的表单控件来说不是一个重要的功能，那就*不要*使用 `type="number"`；此时，请使用带有 [`pattern`](/zh-CN/docs/Web/HTML/Attributes/pattern) 属性的 [`inputmode="numeric"`](/zh-CN/docs/Web/HTML/Global_attributes/inputmode)，将字符限制在数字和相关字符。如果坚持使用 `<input type="number">`，用户在试图做其他事情时，有可能意外地增加一个数字。此外，如果用户试图输入不是数字的东西，也没有明确的反馈说明他们做错了什么。
```

File: files/zh-cn/web/html/element/input/number/index.md [hunk 4]
```diff
@@ -337,8 +422,10 @@ After declaring a few variables, we add an event listener to the button to contr
 
 {{Compat}}
 
-## 相关链接
+## 参见
 
-- [HTML forms guide](/zh-CN/docs/Learn/HTML/Forms)
+- [HTML 表单指南](/zh-CN/docs/Learn/HTML/Forms)
```
Comment (jasonren0403) [inline files/zh-cn/web/html/element/input/number/index.md:427]:
```suggestion
- [HTML 表单指南](/zh-CN/docs/Learn/Forms)
```

File: files/zh-cn/web/html/element/input/number/index.md [hunk 5]
```diff
@@ -3,108 +3,160 @@ title: <input type="number">
 slug: Web/HTML/Element/Input/number
 ---
 
-{{HTMLRef}}
+{{HTMLRef("Input_types")}}
 
-**`"number"`**类型的 {{HTMLElement("input")}} 元素用于让用户输入一个数字。其包括内置验证以拒绝非数字输入。浏览器可能会选择提供步进箭头，让用户可以使用鼠标增加和减少输入的值，或者只需用指尖敲击即可。
+**`"number"`** 类型的 {{HTMLElement("input")}} 元素用于让用户输入一个数字，其包括内置验证以拒绝非数字输入。
```
Comment (yin1999) [inline files/zh-cn/web/html/element/input/number/index.md:8]:
```suggestion
**`number`** 类型的 {{HTMLElement("input")}} 元素用于让用户输入一个数字，其包括内置验证以拒绝非数字输入。
```

File: files/zh-cn/web/html/element/input/number/index.md [hunk 6]
```diff
@@ -3,108 +3,160 @@ title: <input type="number">
...
 
-> **备注：** 不支持 `"number"` 类型的浏览器会改用标准的 [`"text"`](/zh-CN/docs/Web/HTML/Element/input/text) 输入框。
+浏览器可能会选择提供步进箭头，让用户可以使用鼠标增加和减少输入的值，或者只需用指尖敲击即可。
 
-```html
-<input id="number" type="number">
-```
+不支持 `number` 类型的浏览器会回退为标准的 `text` 输入框。
```
Comment (yin1999) [inline files/zh-cn/web/html/element/input/number/index.md:12]: 调整顺序

File: files/zh-cn/web/html/element/input/number/index.md [hunk 7]
```diff
@@ -3,108 +3,160 @@ title: <input type="number">
...
 
-{{ EmbedLiveSample('Basic_example', 600, 40) }}
+{{EmbedInteractiveExample("pages/tabbed/input-number.html", "tabbed-shorter")}}
 
 <table class="properties">
  <tbody>
   <tr>
    <td><strong><a href="#值">值</a></strong></td>
-   <td>表示一个数字的 {{jsxref("Number")}} ，或 空</td>
+   <td>表示一个数字的 {{jsxref("Number")}} ，或为空</td>
```
Comment (yin1999) [inline files/zh-cn/web/html/element/input/number/index.md:20]:
```suggestion
   <td>表示一个数字的 {{jsxref("Number")}}，或为空</td>
```

File: files/zh-cn/web/html/element/input/number/index.md [hunk 8]
```diff
@@ -3,108 +3,160 @@ title: <input type="number">
...
+
+### `max`
+
+允许值范围内的最大值。如果输入到元素中的 {{htmlattrxref("value", "input")}} 超过此值，则元素将无法通过[约束验证](/zh-CN/docs/Web/Guide/HTML/Constraint_validation)。如果 `max` 属性的值不是数字，则元素没有最大值。
+
+此值必须大于或等于 `min` 属性的值。
+
+### `min`
+
+允许值范围内的最小值。如果元素的 {{htmlattrxref("value", "input")}} 小于此值，则该元素将无法通过 [约束验证](/zh-CN/docs/Web/Guide/HTML/Constraint_validation)。如果为 `min` 指定的值不是有效数字，则输入没有最小值。
```
Comment (yin1999) [inline files/zh-cn/web/html/element/input/number/index.md:71]:
```suggestion
允许值范围内的最小值。如果元素的 {{htmlattrxref("value", "input")}} 小于此值，则该元素将无法通过[约束验证](/zh-CN/docs/Web/Guide/HTML/Constraint_validation)。如果为 `min` 指定的值不是有效数字，则输入没有最小值。
```

File: files/zh-cn/web/html/element/input/number/index.md [hunk 9]
```diff
@@ -3,108 +3,160 @@ title: <input type="number">
...
+
+该值必须小于或等于 `max` 属性的值。
+
+### `placeholder`
+
+`placeholder` 属性是一个字符串，可向用户提供有关该字段中需要什么样的信息的简短提示。它应该是说明预期的数据类型的单词或短语，而不是说明性消息。文本中*不得*包含回车符或换行符。
+
+如果控件的内容具有方向（({{Glossary("LTR")}} 或 {{Glossary("RTL")}}），但需要以相反的方向显示占位符，则可以使用 Unicode 双向算法来格式化字符，从而覆盖原有占位符的方向；请参见[如何针对 bidi 文字使用 Unicode 控制符](https://www.w3.org/International/questions/qa-bidi-unicode-controls)获取更多信息。
+
+> **Note:** 请尽可能避免使用 `placeholder` 属性，它在语义上没有其他解释表单的方式有用，并且可能会导致额外的问题。参见 {{SectiononPage("/zh-CN/docs/Web/HTML/Element/input", "标签与占位符")}}以获取更多信息。
```
Comment (yin1999) [inline files/zh-cn/web/html/element/input/number/index.md:81]:
```suggestion
> **备注：** 请尽可能避免使用 `placeholder` 属性，它在语义上没有其他解释表单的方式有用，并且可能会导致额外的问题。参见 {{SectiononPage("/zh-CN/docs/Web/HTML/Element/input", "标签与占位符")}}以获取更多信息。
```

File: files/zh-cn/web/html/element/input/number/index.md [hunk 10]
```diff
@@ -3,108 +3,160 @@ title: <input type="number">
...
+
+允许值范围内的最小值。如果元素的 {{htmlattrxref("value", "input")}} 小于此值，则该元素将无法通过 [约束验证](/zh-CN/docs/Web/Guide/HTML/Constraint_validation)。如果为 `min` 指定的值不是有效数字，则输入没有最小值。
+
+该值必须小于或等于 `max` 属性的值。
+
+### `placeholder`
+
+`placeholder` 属性是一个字符串，可向用户提供有关该字段中需要什么样的信息的简短提示。它应该是说明预期的数据类型的单词或短语，而不是说明性消息。文本中*不得*包含回车符或换行符。
+
+如果控件的内容具有方向（({{Glossary("LTR")}} 或 {{Glossary("RTL")}}），但需要以相反的方向显示占位符，则可以使用 Unicode 双向算法来格式化字符，从而覆盖原有占位符的方向；请参见[如何针对 bidi 文字使用 Unicode 控制符](https://www.w3.org/International/questions/qa-bidi-unicode-controls)获取更多信息。
```
Comment (yin1999) [inline files/zh-cn/web/html/element/input/number/index.md:79]:
```suggestion
如果控件的内容具有方向（{{Glossary("LTR")}} 或 {{Glossary("RTL")}}），但需要以相反的方向显示占位符，则可以使用 Unicode 双向算法来格式化字符，从而覆盖原有占位符的方向；请参见[如何针对双向文本使用 Unicode 控制符](https://www.w3.org/International/questions/qa-bidi-unicode-controls)获取更多信息。
```

File: files/zh-cn/web/html/element/input/number/index.md [hunk 11]
```diff
@@ -3,108 +3,160 @@ title: <input type="number">
...
+
+> **备注：** 由于只读字段不可以拥有值，`required` 不会对指定了 `readonly` 属性的字段起作用。
+
+### `step`
+
+`step` 属性指定了值必须满足的粒度，或者是下文描述的特殊值 `any`。值必须满足基础的步进值，才有效。如果指定了 [`min`](#min) 属性，则由 min 属性决定，否则，使用 {{htmlattrxref("value", "input")}} 属性的值，如果上述两个值都不存在，则提供适当的默认值。
+
+字符串值 `any` 意味着不使用步进值，任意值都可以接受（除其他制约因素如 [`min`](#min) 或 [`max`](#max) 之外）。
+
+> **Note:** 当用户输入的数据不符合步进配置时，{{Glossary("user agent","用户代理")}}可能会四舍五入到最近的有效值，当有两个同样接近的选项时，更倾向于正方向的数字。
```
Comment (yin1999) [inline files/zh-cn/web/html/element/input/number/index.md:95]:
```suggestion
> **备注：** 当用户输入的数据不符合步进配置时，{{Glossary("user agent","用户代理")}}可能会四舍五入到最近的有效值，当有两个同样接近的选项时，更倾向于正方向的数字。
```

**Review by yin1999 — CHANGES_REQUESTED** [review]

**Review by yin1999 — APPROVED** [review]

**Pull request merged**